src/app/portlets/dot-edit-page/content/components/dot-edit-page-workflows-actions/dot-edit-page-workflows-actions.component.html

```diff
@@ -0,0 +1,6 @@
+<p-splitButton 
+    flat 
+    class="edit-page-toolbar__actions"
+    [label]="label"
+    [model]="workflowsActions">
```

Comment:
Why you didn't use async pipe?

Reply:
this is related with the previous one

Reply:
Sure you can, here is how:

Declare it as an Observable
```workflowsActions: Observable<MenuItem[]>;```

Make ```getWorkflowOptions``` to return an MenuItem[]
```typescript
    private getWorkflowOptions(workflows: Workflow[]): MenuItem[] {
        return workflows.map((workflow: Workflow) => {
            return {
                label: workflow.name
            };
        });
    }
```

Then assign the Observable (not sure if need to use map or mergeMap)
```typescript
this.workflowsActions = this.workflowsService
            .getPageWorkflows(this.inode)
            .map((workflows: Workflow[]) => this.getWorkflowOptions(workflows));
```

And then in the HTML use the async pipe

```html
<p-splitButton 
    flat 
    class="edit-page-toolbar__actions"
    [label]="label"
    [model]="workflowsActions | async">
</p-splitButton>
```

And just like that you don't need the: ```pageWorkflows```, also event with no Observable, you don't need it because you could do: 

this.workflowsActions = this.getWorkflowOptions(data);

And don't forget to type the response from the service

```.subscribe((workflows: Workflow[])```

Reply:
Sure you can, here is how:

### 1. Declare it as an Observable
```workflowsActions: Observable<MenuItem[]>;```

### 2. Make ```getWorkflowOptions``` to return an ```MenuItem[]``` and receive ```Workflow[]```
```typescript
    private getWorkflowOptions(workflows: Workflow[]): MenuItem[] {
        return workflows.map((workflow: Workflow) => {
            return {
                label: workflow.name
            };
        });
    }
```

### 3. Then assign the Observable (not sure if need to use map or mergeMap)
```typescript
this.workflowsActions = this.workflowsService
            .getPageWorkflows(this.inode)
            .map((workflows: Workflow[]) => this.getWorkflowOptions(workflows));
```

### 4. And then in the HTML use the async pipe

```html
<p-splitButton 
    flat 
    class="edit-page-toolbar__actions"
    [label]="label"
    [model]="workflowsActions | async">
</p-splitButton>
```

PD.

Also even with no Observable, you don't need it because you could do: 

this.workflowsActions = this.getWorkflowOptions(data);

And don't forget to type the response from the service

```.subscribe((workflows: Workflow[])```